hosts/pds3.py

```diff
@@ -0,0 +1,172 @@
+            lines[i] = cleaner(lines[i])
+
+    return lines
+
+#===========================================================================
+def clean_label(lines, cleaners=None, no_standard=False):
+    """
+    Correct known errors in PDS3 labels.
+
+    Inputs:
+        lines:          List of lines comprising the label.
+        cleaners:       List of functions that take a line as input and
+                        return a cleaned line.  By default, these functions
+                        are appended to a list of standard cleaner functions.
+        no_standard:    If True, the standard cleaners are not used.
+    """
+
+    # Call standard cleaners
+    standard_cleaners = [clean_comments]
+    if not no_standard:
+        lines = call_cleaners(standard_cleaners, lines)
+
+    # Call custom cleaners
+    if cleaners is not None:
+        lines = call_cleaners(cleaners, lines)
+
+    return lines
+
+#===========================================================================
+def find_label(filespec):
+    """Find the PDS3 label corresponding to the given filespec."""
+
+    # Construct candidate label filenames
+    spec = Path(filespec)
+    labelspecs = [
+        spec.with_suffix('.lbl'),
+        spec.with_suffix('.LBL') ]
+    if filespec.upper().endswith('.LBL'):
+        labelspecs.append(filespec)
+
+    # Check candidates; return first one that exists
+    for labelspec in labelspecs:
+        if os.path.isfile(labelspec):
+            return labelspec
+
+    # If no successful candidate, assume attached label
+    return filespec
+
+#===========================================================================
+def get_label(filespec, cleaners=None, no_standard=False):
+    """
+    Find the PDS3 label, clean it, load it, and parse into dictionary.
+
+    Inputs:
+        filespec:       File specification for the label.
+        cleaners:       List of functions that take a label line as input and
+                        return a cleaned line.  By default, these functions
+                        are appended to a list of standard cleaner functions.
+        no_standard:    If True, the standard cleaners are not used.
+    """
+
+    assert os.path.isfile(filespec), f'Not found: {filespec}'
+
+    # Find the label
+    labelspec = find_label(filespec)
+
+    # Load and clean the PDS label
+    lines = clean_label(pdsparser.PdsLabel.load_file(labelspec),
+                        cleaners=cleaners, no_standard=no_standard)
+
+    # Parse the label into a dictionary
+    return pdsparser.PdsLabel.from_string(lines).as_dict()
+
+
+################################################################################
+# UNIT TESTS
+################################################################################
+import unittest
+import os.path
+
+from hosts                   import pds3
+from oops.unittester_support import TESTDATA_PARENT_DIRECTORY
+
+
+#===========================================================================
```

Comment:
I have left these in for now, so that the code can be tested without this pull requiest affecting anything else.  Once we merge this file I'll move the tests to more approriate modules.